libraries/AP_BoardConfig/board_drivers.cpp

```diff
@@ -199,11 +285,34 @@ void AP_BoardConfig::validate_board_type(void)
 #endif
 }
 
+
+
 /*
   auto-detect board type
  */
 void AP_BoardConfig::board_autodetect(void)
 {
+#if defined(HAL_CHIBIOS_ARCH_CUBEBLACK)
+    {
```

Comment:
Nitpick: Why the extra scoping here?

Reply:
Scope is for the success variable.

Reply:
I think adding a board_check_CubeBlack which gets called from here would be better. This won't be the last board where we hit this type of issue

Reply:
Took tridge's suggestion